pkg/controller/gather_commands.go

```diff
@@ -399,6 +400,17 @@ type dataStatus struct {
 	Status    string `json:"status"`
 }
 
+// retryCounter is a helper struct to store number of attempts
+// for each failure type when processing data
+type retryCounter struct {
+	network int
+	request int
+	status  int
+
+	// maximum retry attempts allowed for each failure type before failing completely
+	max int
```

Comment:
As a recommendation for the future, avoid using overly generic variable names. This is because, if a function is lengthy, it's easy to lose focus on what a variable means. The term `maxRetries` is easier to understand than `max` and save you time checking the documentation.

No need to change it right now, as it's a short function and inside a struct. But it's helpful to keep this in mind for future coding.